### PR TITLE
feat: Taxes effect is added

### PR DESCRIPTION
**Pull request by asternotus**

### CODE REVIEW

File: effects/taxes_effect.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    def _validate(self, target_number: int) -> bool:
+        return utils.validate_citizen_target_number(target_number,
+                                                    self.game.citizens)
+
+    def _on_clear_impl(self) -> None:
+        pass
+
+    def _choose_payment(self) -> bool:
+        taxes_choice = self.user_interaction.read_number(
+            "Вас посетил Соцработник. Выберите вариант оплаты налогов:\n1. Потратить 1 HP\n2. Потратить карту\n"
```
Comment (elias-gvin) [inline effects/taxes_effect.py]: Don't use raw strings. Add it to message_text_config module

File: effects/taxes_effect.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        elif taxes_choice == 2 and self.targets[
+                0].citizen_card is not None or len(
+                    self.targets[0].stolen_cards) > 0:
+            while (True):
+                if self._choose_card_payment():
+                    break
+            return True
+        else:
+            self.user_interaction.show_active_instant(
+                "Введите корректное значение")
```
Comment (elias-gvin) [inline effects/taxes_effect.py]: Don't use raw strings. Use values from message_text_config module

File: effects/taxes_effect.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                                                    self.game.citizens)
+
+    def _on_clear_impl(self) -> None:
+        pass
+
+    def _choose_payment(self) -> bool:
+        taxes_choice = self.user_interaction.read_number(
+            "Вас посетил Соцработник. Выберите вариант оплаты налогов:\n1. Потратить 1 HP\n2. Потратить карту\n"
+        )
+        if taxes_choice == 1:
```
Comment (elias-gvin) [inline effects/taxes_effect.py]: separate game logic and interaction with a user.
_choose_payment should either read values options user OR modify user's HP and cards.
I think that _choose_payment method should ONLY return players chosen option (you can create enum for this with two values) to resolve_impl

File: effects/taxes_effect.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+        return True
+
+    def _resolve_impl(self) -> bool:
+        if isinstance(self.targets[0], Player):
+            if self.activation_round == self.game.round_number:
+                return False
+            else:
+                while (True):
+                    if self._choose_payment():
```
Comment (elias-gvin) [inline effects/taxes_effect.py]: You can move infinit cycle to _choose_payment method

File: effects/taxes_effect.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+                        break
+
+        elif isinstance(self.targets[0], Spy):
```
Comment (elias-gvin) [inline effects/taxes_effect.py]: Remove this.
All effects interactions with Spy would be made inside Spy effect

**Review by elias-gvin — CHANGES_REQUESTED** [review]

File: effects/taxes_effect.py (hunk 1 above)
Comment (asternotus) [reply effects/taxes_effect.py]: Done

File: effects/taxes_effect.py (hunk 2 above)
Comment (asternotus) [reply effects/taxes_effect.py]: Done

File: effects/taxes_effect.py (hunk 5 above)
Comment (asternotus) [reply effects/taxes_effect.py]: Done

File: effects/taxes_effect.py (hunk 4 above)
Comment (asternotus) [reply effects/taxes_effect.py]: Done

File: effects/taxes_effect.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+            else:
+                payment_choice = self._choose_payment()
+                if payment_choice == PaymentChoice.HP_CHOICE:
+                    self.targets[0].hp -= 1
+                elif payment_choice == PaymentChoice.CARD_CHOICE and (
+                        self.targets[0].citizen_card is not None
+                        or len(self.targets[0].stolen_cards) > 0):
+                    self._choose_card_payment()
+        else:
+            self.targets[0].citizen_card = None
```
Comment (elias-gvin) [inline effects/taxes_effect.py:48]: If citizen have no cards - he should receive 1 strike
(i'll add this logic. Just a reminder for myself)

**Review by elias-gvin — CHANGES_REQUESTED** [review]

**Review by elias-gvin — APPROVED** [review]

**Pull request merged**